packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -1889,9 +1889,14 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
         oldRoute
           ..didComplete(result ?? oldRoute.currentResult)
           ..dispose();
+        final int finalIndex = _history.indexOf(newRoute);
+        if (finalIndex > 0) {
```

Comment:
This change seems to come back from the revert

Reply:
we are double notifying the new previous route, is this necessary given we already notifying pre transition in line 1902? Is there an issue depends on this change?